playbooks/ansible-cloud/k8s/pre.yaml

```diff
@@ -0,0 +1,50 @@
+---
+- hosts: controller
+  tasks:
+    - name: Ensure docker is running
+      include_role:
+        name: ensure-docker
+
+    - name: Create python virtual env
+      command: python3 -m venv '{{ ansible_test_venv_path }}'
+      args:
+        creates: '{{ ansible_test_venv_path }}'
+
+    - name: create collection directory
+      ansible.builtin.file:
+        path: '{{ item }}'
+        state: directory
+      with_items:
+        - ~/.ansible/collections/ansible_collections/kubernetes
+        - ~/.ansible/collections/ansible_collections/cloud
+
+    - name: Install collections using symlink
+      ansible.builtin.file:
+        state: link
+        path: '{{ item.dest }}'
+        src: '{{ ansible_user_dir }}/{{ zuul.projects[item.src].src_dir }}'
+      with_items:
+        - src: github.com/ansible-collections/kubernetes.core
+          dest: ~/.ansible/collections/ansible_collections/kubernetes/core
+        - src: github.com/ansible-collections/cloud.common
+          dest: ~/.ansible/collections/ansible_collections/cloud/common
```

Comment:
we have a role to do this called deploy-artifacts: https://github.com/ansible/ansible-zuul-jobs/blob/master/roles/deploy-artifacts/tasks/main.yaml

You can use that, along with the build-ansible-collection job. You can look to other network jobs to see how it works.

Reply:
This role is running ``ansible-galaxy collection install (collection)`` which will install collection based on ``galaxy.yml`` or ``Manifest.json`` which is not what I intend, I am trying to use the ``master`` branch of the collection

Reply:
This is not entirely true, we build all collections ourself in the build-ansible-collection job, which will pull content from git rather then galaxy. We then, install the collection via a local tarball.

Basically, it will give you the same you are doing here, expect using a standard way of installing content.

Reply:
ok I will then use this, thanks